File: 0046-permutations/0046-permutations.java
class Solution {
    public List<List<Integer>> permute(int[] nums) {
        List<List<Integer>> answer = new ArrayList<>();
        backTrack(nums,new ArrayList<>(),answer);
        return answer;
    }
    public void backTrack(int[] nums, List<Integer> comb, List<List<Integer>> answer) {
        if(comb.size() == nums.length) {
            answer.add(new ArrayList<>(comb));
        }else {
            for(int i=0;i<nums.length;i++) {
                if(comb.contains(nums[i])) continue;
                comb.add(nums[i]);
                backTrack(nums,comb,answer);
                comb.remove(comb.size()-1);
            }
        }
    }
}

File: 0046-permutations/0046-permutations.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        res = []
        def backtrack(curr: List[int]):
            if len(curr) == len(nums):
                res.append(curr.copy())
                return
            
            for n in nums:
                if n not in curr:
                    curr.append(n)
                    backtrack(curr)
                    curr.pop()

        backtrack([])

        return res